feat: produce code in two languages to determine prime numbers

NcoleJac opened the pull request:
I am in the process of producing code in two languages that will determine if a number is prime.  
After watching the videos, I determined to go about the assignment by describing 3 logic statements describing a process that I would use without a computer.  Afterwards, I used those statements along with the lesson prompt to get assistance from Chatgpt.  
Problems I encountered: 
Based on the information I provided, I was able to successfully produce in python a code capable of determining prime numbers up to 20.  
Regarding the JavaScript, I was not able to check my code yet. 
My goal is to improve the python code to manage all numbers greater than 2.  With JavaScript I intend to do the same and run a test as well.

anthonydmays reviewed (commented):
## 🎓 Automated Grading Report

**Student:** Nicole Jackson
**Date:** 8/16/2025

### Feedback
The submission includes implementations of the `isPrime` function in both Python and JavaScript, as required. However, there are some issues with the implementations. In the Python code, the loop condition `range(2, n)` is correct, but the `if divisor == n: continue` line is unnecessary since the loop never reaches `n`. In the JavaScript code, the loop condition `for (let divisor = 2; divisor <= 9; divisor++)` is incorrect because it only checks divisors up to 9, which is insufficient for numbers greater than 9. The explanation provided in the README.md file is clear, but it could be improved by addressing the logic differences in the loop conditions. The technical requirement of having a README.md file is met, but the inclusion of separate code files was not necessary and could have been consolidated into the README.md. No unit tests were provided, so the stretch criteria were not attempted. Overall, the submission demonstrates a good effort but requires some corrections to fully meet the functional and technical requirements.

---
*This is an automated preliminary review. Please review and adjust before finalizing.*

NcoleJac commented:
Initially when I wrote both the Python and JavaScript codes I used the same loop to check for divisors between 2 and 9. I added that based on the way I used to think about finding greatest common factors back in school. I thought every number would have at least one divisor less than or equal to 9, so it made sense to me at the time.

I didn’t realize that the 2–9 loop doesn’t actually work for checking primes. Can you help me understand why? In addition, I thought I had updated both versions to use that loop, but it looks like I missed it in one of them.   And I see how my README does not reflect this difference between the two languages.  Should I make a couple of edits?

I read an explanation on chatgpt that explained my Python code would take longer to run because it was written to check every single number up to one less than the input. 
Which code is better, in your opinion?

NcoleJac commented:
*every natural number

anthonydmays commented on lesson_04/nicolejackson/README.md:
```diff
@@ -0,0 +1,60 @@
+        return false;
+    }
+
+    for (let divisor = 2; divisor <= 9; divisor++) {
```
This loop isn't going to work correctly for solving the problem.
